src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -669,7 +1160,8 @@ class spell_paletress_summon_memory : public SpellScriptLoader
 
             void HandleScript(SpellEffIndex /*effIndex*/)
             {
-                GetHitUnit()->CastSpell(GetHitUnit(), memorySpellId[urand(0, 24)], GetCaster()->GetGUID());
+                uint32 const randMemorySpellId = Trinity::Containers::SelectRandomContainerElement(memorySpells);
+                GetHitUnit()->CastSpell(GetHitUnit(), randMemorySpellId);
```

Comment:
I believe this behaviour is incorrect, the memory is rolled once per instance and then the same memory is called upon again, in case of a wipe for example.